PicoPagesList.php

```diff
@@ -55,8 +66,30 @@ public function onPagesLoaded(
      */
     public function onPageRendering(Twig_Environment &$twig, array &$twigVariables, &$templateName)
     {
-        $twigVariables['PagesList'] = $this;
-        $twigVariables['pages_list'] = $this->html(); // backward compatibility
+        $twig->addGlobal('nested_pages', $this->items);
+        $twig->addFilter(new Twig_SimpleFilter('navigation', function($pages) {
+            return $this->output($pages);
```

Comment:
Noted !
For my education, what exactly is requiring 5.4+ here ? And is there any analysis tool/workflow, apart being experienced, to have a global report on requirements ?
I see that `[]` for example, that I used recently without thinking about it, is 5.4. It wouldn't be a hassle to use the old `array()` instead, but it would be useless if I don't know the requirements of every other pieces and language features.

Reply:
You can't use `$this` in a closure with PHP 5.3. Binding a Closure to a context (i.e. `$this`, what is also the default since then) was added in PHP 5.4. I personally don't know any tool that can check a project's compatibility with different PHP versions. PHP usually maintains BC and, at best, removes features that were discouraged for a long time, so you basically just have to know when a new feature has been introduced (see e.g. https://secure.php.net/manual/en/migration54.php)

Reply:
Ok, thanks. That's good to know.
In this case here I can remove the closure and the array shortcuts, but I'll not be able to test further and validate 5.3 support.

```php
$twig->addFilter(new Twig_SimpleFilter('navigation', array(this, 'output')));
```